runtime/lua/vim/_opt.lua

```diff
@@ -349,8 +349,16 @@ local create_option_metatable = function(set_local)
         vim.api.nvim_set_option(self._name, value)
       elseif is_buffer_option(info) then
         vim.api.nvim_buf_set_option(0, self._name, value)
+
+        if not set_local and not info.global_local then
+          vim.api.nvim_set_option(self._name, value)
+        end
       elseif is_window_option(info) then
         vim.api.nvim_win_set_option(0, self._name, value)
+
+        if not set_local and not info.global_local then
```

Comment:
IIRC only needed for buf options, not window options (as windows, unlike buffers, proliferate by subdivision)

Reply:
What's a good way to test this? I'd like to write a functional test for this... but tbh I don't even know if I 100% understand the behavior.

You think just setting the option, `:vsplit` and then see the options in new buffer?